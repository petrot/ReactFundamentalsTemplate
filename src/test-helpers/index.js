export * from "./local-storage.mock";
export * from "./store.mock";
export * from "./test-wrapper.mock";
