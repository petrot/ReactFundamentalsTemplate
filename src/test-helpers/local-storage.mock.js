export const mockGetItem = jest.fn();
export const mockSetItem = jest.fn();
export const mockRemoveItem = jest.fn();

export const prepareMockLocalStorage = () => {
  Object.defineProperty(window, "localStorage", {
    value: {
      getItem: (...args) => mockGetItem(...args),
      setItem: (...args) => mockSetItem(...args),
      removeItem: (...args) => mockRemoveItem(...args),
    },
  });
};
