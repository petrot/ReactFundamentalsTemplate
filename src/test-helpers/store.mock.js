export const TEST_NAME = "Test Name";

export const mockedState = {
  user: {
    isAuth: true,
    name: TEST_NAME,
    role: "admin",
    token: "mock-token",
  },
  courses: [
    {
      title: "Course 1",
      description: "Description 1",
      duration: 5,
      authors: [
        "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
        "1c972c52-3198-4098-b6f7-799b45903199",
        "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
      ],
      creationDate: "11/03/2024",
      id: "b28fd9ad-c683-4900-bf99-38a1c5748309",
    },
  ],
  authors: [
    {
      name: "author",
      id: "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
    },
    {
      name: "author2",
      id: "1c972c52-3198-4098-b6f7-799b45903199",
    },
    {
      name: "author3",
      id: "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    },
  ],
};

export const mockedStore = {
  getState: () => mockedState,
  subscribe: jest.fn(),
  dispatch: jest.fn(),
};
