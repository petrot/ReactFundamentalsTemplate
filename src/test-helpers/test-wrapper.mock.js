import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import { mockedStore } from "./store.mock";

export const TestWrapper = ({ children, store = mockedStore }) => (
  <Provider store={store}>
    <Router>{children}</Router>
  </Provider>
);
