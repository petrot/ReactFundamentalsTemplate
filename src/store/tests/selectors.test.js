import { mockedState } from "../../test-helpers";
import {
  getAuthorsSelector,
  getCoursesSelector,
  getUserNameSelector,
  getUserRoleSelector,
  getUserTokenSelector,
} from "../selectors";

describe("Selectors", () => {
  it("should get courses", async () => {
    const courses = getCoursesSelector(mockedState);

    expect(courses).toMatchSnapshot();
  });

  it("should get authors", async () => {
    const authors = getAuthorsSelector(mockedState);

    expect(authors).toMatchSnapshot();
  });

  it("should get user name", async () => {
    const userName = getUserNameSelector(mockedState);

    expect(userName).toMatchInlineSnapshot(`"Test Name"`);
  });

  it("should get user role", async () => {
    const userRole = getUserRoleSelector(mockedState);

    expect(userRole).toMatchInlineSnapshot(`"admin"`);
  });

  it("should get user token", async () => {
    const userToken = getUserTokenSelector(mockedState);

    expect(userToken).toMatchInlineSnapshot(`"mock-token"`);
  });
});
