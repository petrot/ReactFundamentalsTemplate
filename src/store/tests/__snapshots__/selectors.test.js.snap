// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectors should get authors 1`] = `
Array [
  Object {
    "id": "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
    "name": "author",
  },
  Object {
    "id": "1c972c52-3198-4098-b6f7-799b45903199",
    "name": "author2",
  },
  Object {
    "id": "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    "name": "author3",
  },
]
`;

exports[`Selectors should get courses 1`] = `
Array [
  Object {
    "authors": Array [
      "9b87e8b8-6ba5-40fc-a439-c4e30a373d36",
      "1c972c52-3198-4098-b6f7-799b45903199",
      "072fe3fc-e751-4745-9af5-aa9eed0ea9ed",
    ],
    "creationDate": "11/03/2024",
    "description": "Description 1",
    "duration": 5,
    "id": "b28fd9ad-c683-4900-bf99-38a1c5748309",
    "title": "Course 1",
  },
]
`;
